code/modules/pool/pool_structures.dm

```diff
@@ -0,0 +1,151 @@
+/obj/structure/pool
+	name = "pool"
+	icon = 'icons/obj/machines/pool.dmi'
+	anchored = TRUE
+	resistance_flags = UNACIDABLE|INDESTRUCTIBLE
+
+/obj/structure/pool/ladder
+	name = "Ladder"
+	icon_state = "ladder"
+	desc = "Are you getting in or are you getting out?."
+	layer = ABOVE_MOB_LAYER
+	dir = EAST
+
+/obj/structure/pool/ladder/attack_hand(mob/living/user as mob)
+	if(Adjacent(user) && user.y == y && user.swimming == 0)
+		user.swimming = TRUE
+		user.forceMove(get_step(user, get_dir(user, src))) //Either way, you're getting IN or OUT of the pool.
+	else if(user.loc == loc && user.swimming == TRUE)
+		user.swimming = FALSE
+		user.forceMove(get_step(user, turn(dir, 180)))
+
+/obj/structure/pool/Rboard
+	name = "JumpBoard"
+	density = FALSE
+	icon_state = "boardright"
+	desc = "The less-loved portion of the jumping board."
+	dir = EAST
+
+/obj/structure/pool/Rboard/CheckExit(atom/movable/O as mob|obj, target as turf)
+	if(istype(O) && O.pass_flags & PASSGLASS)
+		return TRUE
+	if(get_dir(O.loc, target) == dir)
+		return FALSE
+	return TRUE
+
+/obj/structure/pool/Lboard
+	name = "JumpBoard"
+	icon_state = "boardleft"
+	desc = "Get on there to jump!"
+	layer = FLY_LAYER
+	dir = WEST
+	var/jumping = FALSE
+	var/timer
+
+/obj/structure/pool/Lboard/proc/backswim(obj/O, mob/living/user) //Puts the sprite back to it's maiden condition after a jump.
+	if(jumping)
+		for(var/mob/living/jumpee in loc) //hackzors.
+			playsound(jumpee, 'sound/effects/splash.ogg', 60, TRUE, 1)
+			jumpee.layer = 4
+			jumpee.pixel_x = 0
+			jumpee.pixel_y = 0
+			jumpee.Stun(2)
+			jumpee.swimming = TRUE
+
+/obj/structure/pool/Lboard/attack_hand(mob/living/user)
+	if(iscarbon(user))
+		var/mob/living/carbon/jumper = user
+		if(jumping)
+			to_chat(user, "<span class='notice'>Someone else is already making a jump!</span>")
+			return
+		var/turf/T = get_turf(src)
+		if(user.swimming)
+			return
+		else
+			for(var/obj/machinery/pool/controller/pc in range(4,src)) //Clunky as fuck I know.
+				if(pc.drained)
+					to_chat(user, "<span class='notice'>That would be suicide</span>") //TODO: make this a suicide action.
```

Comment:
forgot a punctuation mark

Reply:
The Todo is mine actually, planning to rewrite it so you can dive into an empty pool and it gives you a warning pop up and does something special and funny.